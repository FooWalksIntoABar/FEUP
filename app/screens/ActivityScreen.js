import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { setActivityStatus, nextActivity } from '../actions/gameActions'
import { addStars } from '../actions/childActions'
import { Image, Text, View, StatusBar, BackHandler } from 'react-native'

import { ProgressBar } from '../components/Activity/ProgressBar'
import { ProgressClock } from '../components/Activity/ProgressClock'
import { CompleteButton } from '../components/Activity/CompleteButton'
import { PauseButton } from '../components/Activity/PauseButton'
import { CancelButton } from '../components/Activity/CancelButton'
import { RewardsModal } from '../components/RewardsModal/RewardsModal'
import Images from '../assets/images/images'

import styles from '../styles/Activity.style'

class ActivityScreen extends Component {
  constructor (props) {
    super(props)

    this.state = {
      elapsedTime: 0,
      progressType: '',
      isPhoto: this.props.activity.photo !== undefined,
      updateRate: 100, // ms
      isPaused: false,
      isCompleted: false,
      isCompletable: false
    }

    this.pauseActivity = this.pauseActivity.bind(this)
    this.cancelActivity = this.cancelActivity.bind(this)
    this.completeActivity = this.completeActivity.bind(this)
    this.resumeActivity = this.resumeActivity.bind(this)
    this.nextActivity = this.nextActivity.bind(this)
    this.backToMenu = this.backToMenu.bind(this)
  }

  componentDidMount () {
    BackHandler.addEventListener('hardwareBackPres', this.cancelActivity)
    this.interval = setInterval(() => {
      if (this.state.isPaused) return
      if (this.state.elapsedTime >= this.props.activity.time.max) this.completeActivity()
      if (this.state.elapsedTime >= this.props.activity.time.min) {
        this.state.isCompletable = true
      }

      this.setState(() => {
        return { elapsedTime: this.state.elapsedTime + this.state.updateRate / 1000 }
      })
    }, this.state.updateRate)
  }

  componentWillUnmount () {
    BackHandler.removeEventListener('hardwareBackPress', this.cancelActivity)
    clearInterval(this.interval)
  }

  pauseActivity () {
    this.setState(() => ({ isPaused: true }))
  }

  cancelActivity () {
    this.props.navigation.popToTop()
    return true // To be called by andrdid back button
  }

  completeActivity () {
    clearInterval(this.interval)
    const status = {
      completed: true,
      reward: this.state.elapsedTime > this.props.activity.time.max ? 0
        : this.state.elapsedTime < this.props.activity.time.goal ? 3
          : this.state.elapsedTime < this.props.activity.time.goal + (this.props.activity.time.max - this.props.activity.time.goal) / 2 ? 2 : 1,
      time: parseInt(this.state.elapsedTime)
    }
    this.props.setActivityStatus(this.props.activity, status)
    this.setState(() => { return { isCompleted: true } })
  }

  nextActivity () {
    if (this.props.activity.status) this.props.addStars(this.props.activity.status.reward)
    this.props.nextActivity()
    this.props.navigation.replace('Activity')
  }

  backToMenu () {
    if (this.props.activity.status) this.props.addStars(this.props.activity.status.reward)
    this.props.navigation.popToTop()
  }

  resumeActivity () {
    this.setState(() => ({ isPaused: false }))
  }

  returnURIorImage = () => {
    // TODO: This checks whether the photo attribute is type URI and should probably just eventually be totally changed to URI.
    if (this.props.activity.photo !== undefined && this.props.activity.photo.includes('file://')) {
      return { uri: this.props.activity.photo }
    } else {
      return Images[this.state.isPhoto ? this.props.activity.photo : this.props.activity.image]
    }
  }

  render () {
    return (
      <View style={[{ backgroundColor: this.props.activity.color }, styles.activityScreen]} >
        <StatusBar hidden />
        <Image
          style={this.state.isPhoto ? styles.photo : styles.image}
          resizeMode={this.state.isPhoto ? 'cover' : 'center'}
          source={this.returnURIorImage()} />
        <View style={styles.titleContainer}>
          <Text style={this.state.isPhoto ? styles.photoTitle : styles.title}>{this.props.activity.title}</Text>
        </View>
        {this.props.progressType === 'clock' && !this.state.isCompleted && <ProgressClock showTimer={this.props.showTimer} elapsedTime={this.state.elapsedTime} activityTimes={this.props.activity.time} isPaused={this.state.isPaused} />}
        {!this.state.isCompleted && <View style={styles.buttonContainer}>
          {this.props.progressType === 'bar' && <ProgressBar showTimer={this.props.showTimer} elapsedTime={this.state.elapsedTime} activityTimes={this.props.activity.time} isPaused={this.state.isPaused} />}
          <CancelButton style={styles.smallButton} cancelActivity={this.cancelActivity} />
          <PauseButton style={styles.smallButton} pauseActivity={this.pauseActivity} resumeActivity={this.resumeActivity} isPaused={this.state.isPaused} />
          <CompleteButton style={styles.largeButton} isCompletable={this.state.isCompletable} completeActivity={this.completeActivity} />
        </View>}
        <RewardsModal
          currentActivity={this.props.currentActivity}
          activities={this.props.activities}
          nextPress={this.nextActivity}
          backPress={this.backToMenu} />
      </View>
    )
  }
}

export default connect(
  state => ({
    progressType: state.settings.activityProgressType,
    showTimer: state.settings.activityShowTimer,
    activity: state.game.routines[state.game.currentRoutine].activities[state.game.currentActivity],
    activities: state.game.routines[state.game.currentRoutine].activities,
    currentActivity: state.game.currentActivity
  }),
  dispatch => ({
    setActivityStatus: (activity, status) => dispatch(setActivityStatus(activity, status)),
    nextActivity: () => dispatch(nextActivity()),
    addStars: stars => dispatch(addStars(stars))
  })
)(ActivityScreen)

ActivityScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  progressType: PropTypes.string.isRequired,
  showTimer: PropTypes.bool.isRequired,
  currentActivity: PropTypes.number.isRequired,
  activities: PropTypes.array.isRequired,
  activity: PropTypes.object.isRequired,
  setActivityStatus: PropTypes.func.isRequired,
  nextActivity: PropTypes.func.isRequired,
  addStars: PropTypes.func.isRequired
}
