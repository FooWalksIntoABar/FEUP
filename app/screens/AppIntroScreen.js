import React, { Component } from 'react'
import AppIntroSlider from 'react-native-app-intro-slider'
import PropTypes from 'prop-types'
import Images from '../assets/images/images'

const slides = [
  {
    key: '1',
    title: 'Texto explicativo #1',
    text: 'A ser preenchido pelas nossas fantásticas product owners.\nO número de páginas e imagem é ajustável consoante necessário.',
    image: Images.socks,
    imageStyle: { width: 240, height: 240 },
    backgroundColor: '#59b2ab'
  },
  {
    key: '2',
    title: 'Texto explicativo #2',
    text: 'A ser preenchido pelas nossas fantásticas product owners.\nO número de páginas e imagem é ajustável consoante necessário.',
    image: Images.socks,
    imageStyle: { width: 240, height: 240 },
    backgroundColor: '#febe29'
  },
  {
    key: '3',
    title: 'Texto explicativo #3',
    text: 'A ser preenchido pelas nossas fantásticas product owners.\nO número de páginas e imagem é ajustável consoante necessário.',
    image: Images.socks,
    imageStyle: { width: 240, height: 240 },
    backgroundColor: '#22bcb5'
  }
]

export default class AppIntroScreen extends Component {
  _onDone = () => {
    this.props.navigation.goBack()
  }

  render () {
    return (
      <AppIntroSlider slides={slides} onDone={this._onDone} />
    )
  }
}

AppIntroScreen.propTypes = {
  navigation: PropTypes.object.isRequired
}
