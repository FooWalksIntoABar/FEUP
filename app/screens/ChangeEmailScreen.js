import React from 'react'
import { View, Text, TextInput, TouchableOpacity } from 'react-native'

import PropTypes from 'prop-types'

import styles from '../styles/ParentStyles/ChangeEmailScreen.style'

export default class ChangeEmailScreen extends React.Component {
  static navigationOptions = {
    headerMode: 'screen',
    navigationOptions: {
      headerTitle: 'Repor E-mail'
    }
  };
  render () {
    return (
      <View style={styles.generalLayout}>
        <View style={styles.container}>
          <Text>E-mail Atual</Text>
          <Text>{this.props.navigation.state.params.email}</Text>
        </View>
        <View style={styles.container}>
          <Text>Novo E-mail</Text>
          <TextInput
            keyboardType='email-address'
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View style={styles.container}>
          <Text>Password</Text>
          <Text>Insira a sua password para confirmar a alteração de e-mail.</Text>
          <TextInput
            secureTextEntry
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View >
          <TouchableOpacity style={styles.confirmBtn} onPress={() => { this.props.navigation.state.params.setEmail('hahahha') }}>
            <Text>Confirmar</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}

ChangeEmailScreen.propTypes = {
  navigation: PropTypes.object.isRequired
}
