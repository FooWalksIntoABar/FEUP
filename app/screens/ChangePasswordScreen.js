import React from 'react'
import { View, Text, TextInput, TouchableOpacity } from 'react-native'

import styles from '../styles/ParentStyles/ChangePasswordScreen.style'

export default class ChangePasswordScreen extends React.Component {
  static navigationOptions = {
    headerMode: 'screen',
    navigationOptions: {
      headerTitle: 'Repor E-mail'
    }
  };
  render () {
    return (
      <View style={styles.generalLayout}>
        <View style={styles.container}>
          <Text>Insira a sua antiga password.</Text>
          <TextInput
            secureTextEntry
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View style={styles.container}>
          <Text>Insira a sua nova password.</Text>
          <TextInput
            secureTextEntry
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View style={styles.container}>
          <Text>Repita a sua nova password.</Text>
          <TextInput
            secureTextEntry
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View>
          <TouchableOpacity onPress={() => { console.log('changed password') }}>
            <Text>Alterar Password</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}
