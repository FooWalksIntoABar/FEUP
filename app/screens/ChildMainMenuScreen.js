import React, { Component } from 'react'
import { View, StatusBar, Image, TouchableOpacity } from 'react-native'
import { ScreenOrientation } from 'expo'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { ChildExperienceBar } from '../components/MainMenu/ChildExperienceBar'

import Images from '../assets/images/images'

import styles from '../styles/ChildMainMenuScreen.style'

class ChildMainMenuScreen extends Component {
  constructor (props) {
    super(props)
    this.openShop = this.openShop.bind(this)
    this.closeShop = this.closeShop.bind(this)
  }

  state = {
    isShopVisible: false
  }

  componentDidMount () {
    ScreenOrientation.allow(ScreenOrientation.Orientation.LANDSCAPE)
  }

  componentWillUnmount () {
    ScreenOrientation.allow(ScreenOrientation.Orientation.PORTRAIT)
  }

  openShop () {
    this.setState({ isShopVisible: true })
  }

  closeShop () {
    this.setState({ isShopVisible: false })
  }

  render () {
    return (
      <View style={styles.mainMenuContainer}>
        <StatusBar hidden />
        <View style={styles.experienceBarContainer}>
          <ChildExperienceBar
            progress={(this.props.xp - this.props.level * 100) / 100}
            level={this.props.level} />
        </View>
        <View style={styles.buttonContainer}>
          <TouchableOpacity
            style={styles.profileButton}
            onPress={() => { this.props.navigation.navigate('ShopScreen') }}
            activeOpacity={0.9} >
            <Image style={styles.profileImage} resizeMode={'contain'} source={Images.avatar} />
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.button}
            onPress={() => { this.props.navigation.navigate('ChooseRoutineScreen') }}
            activeOpacity={0.9}>
            <Image style={styles.buttonImage} source={Images.ui.play} />
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}

export default connect(
  state => ({
    level: state.child.level,
    xp: state.child.xp
  })
)(ChildMainMenuScreen)

ChildMainMenuScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  level: PropTypes.number.isRequired,
  xp: PropTypes.number.isRequired
}
