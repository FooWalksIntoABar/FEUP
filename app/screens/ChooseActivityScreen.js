import React from 'react'
import { StatusBar, View } from 'react-native'
import { connect } from 'react-redux'
import { setCurrentActivity } from '../actions/gameActions'
import PropTypes from 'prop-types'

import CardCarousel from '../components/Carousel/CardCarousel'
import { backgroundColor } from '../styles/Colors'

export class ChooseActivityScreen extends React.Component {
  constructor (props) {
    super(props)
    this.onPress = this.onPress.bind(this)
  }

  onPress = (activity) => {
    this.props.setCurrentActivity(activity)
    this.props.navigation.navigate('Activity')
  }

  render () {
    return (
      <View style={{ backgroundColor: backgroundColor }}>
        <StatusBar hidden />
        <CardCarousel
          data={this.props.activities.filter(activity => activity.status === undefined)}
          onPress={this.onPress} />
      </View>
    )
  }
}

export default connect(
  state => ({
    activities: state.game.routines[state.game.currentRoutine].activities
  }),
  dispatch => ({
    setCurrentActivity: routine => dispatch(setCurrentActivity(routine))
  })
)(ChooseActivityScreen)

ChooseActivityScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  activities: PropTypes.array.isRequired,
  setCurrentActivity: PropTypes.func.isRequired
}
