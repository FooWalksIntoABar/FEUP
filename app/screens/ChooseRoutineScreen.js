import React from 'react'
import { StatusBar, View, Text } from 'react-native'
import { connect } from 'react-redux'
import { setCurrentRoutine, nextActivity } from '../actions/gameActions'
import PropTypes from 'prop-types'

import CardCarousel from '../components/Carousel/CardCarousel'
import styles from '../styles/CardCarousel.style'

class ChooseRoutineScreen extends React.Component {
  constructor (props) {
    super(props)
    this.chooseActivities = this.chooseActivities.bind(this)
    this.startRoutine = this.startRoutine.bind(this)
  }

  static navigationOptions = {
    header: null
  }

  chooseActivities = (routine) => {
    this.props.setCurrentRoutine(routine)
    this.props.navigation.navigate('ChooseActivityScreen')
  }

  startRoutine = (routine) => {
    this.props.setCurrentRoutine(routine)
    this.props.nextActivity()
    this.props.navigation.navigate('Activity')
  }

  render () {
    if (!this.props.routines.every(routine => routine.activities.every(activity => activity.status !== undefined))) {
      return (
        <View style={styles.screenBackground}>
          <StatusBar hidden />
          <CardCarousel
            data={this.props.routines.filter(routine => !routine.activities.every(activity => activity.status !== undefined))}
            onPress={this.props.routinePlayType === 'auto' ? this.startRoutine : this.chooseActivities}
            onButtonPress={this.props.routinePlayType === 'choose' ? this.startRoutine : undefined}
            isRoutine
          />
        </View>
      )
    } else {
      return (<View style={[styles.screenBackground, { justifyContent: 'center', alignItems: 'center' }]}>
        <Text style={styles.tooltip}>Sem rotinas para jogar.{'\n'}Volta amanh√£!</Text>
      </View>)
    }
  }
}

export default connect(
  state => ({
    routines: state.game.routines,
    routinePlayType: state.settings.routinePlayType
  }),
  dispatch => ({
    setCurrentRoutine: routine => dispatch(setCurrentRoutine(routine)),
    nextActivity: () => dispatch(nextActivity())
  })
)(ChooseRoutineScreen)

ChooseRoutineScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  routinePlayType: PropTypes.string.isRequired,
  routines: PropTypes.array.isRequired,
  setCurrentRoutine: PropTypes.func.isRequired,
  nextActivity: PropTypes.func.isRequired
}
