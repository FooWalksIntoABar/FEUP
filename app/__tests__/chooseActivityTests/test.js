import 'react-native'
import React from 'react'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { ChooseActivityScreen } from '../../screens/ChooseActivityScreen'

configure({ adapter: new Adapter() })

describe('ChooseActivityScreen snapshot', () => {
  const activities = [
    {
      title: 'Fazer a cama',
      image: 'bed',
      photo: 'bedroom',
      color: '#7d84b2',
      time: {
        min: 0,
        max: 120,
        goal: 60
      }
    },
    {
      title: 'Lavar os dentes',
      image: 'toothbrush',
      color: '#0e79b2',
      time: {
        min: 1,
        max: 15,
        goal: 10
      }
    },
    {
      title: 'Vestir',
      image: 'socks',
      color: '#7fb800',
      time: {
        min: 10,
        max: 120,
        goal: 60
      }
    },
    {
      title: 'Tomar banho',
      image: 'shower',
      color: '#37c1f0',
      time: {
        min: 10,
        max: 120,
        goal: 60
      }
    },
    {
      title: 'Preparar a mochila',
      image: 'bag',
      color: '#e43f6f',
      time: {
        min: 10,
        max: 120,
        goal: 60
      }
    },
    {
      title: 'Calçar os sapatos',
      image: 'sneakers',
      color: '#4bb3fd',
      time: {
        min: 10,
        max: 120,
        goal: 60
      }
    },
    {
      title: 'Tomar o pequeno almoço',
      image: 'breakfast',
      color: '#ff7f11',
      time: {
        min: 10,
        max: 120,
        goal: 60
      }
    },
    {
      title: 'Pentear cabelo',
      image: 'comb',
      color: '#b0db43',
      time: {
        min: 10,
        max: 120,
        goal: 60
      }
    }
  ]

  it('renders ChooseActivityScreen correctly', () => {
    const wrapper = shallow(<ChooseActivityScreen
      navigation={{}}
      activities={activities}
      setCurrentActivity={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })
})
