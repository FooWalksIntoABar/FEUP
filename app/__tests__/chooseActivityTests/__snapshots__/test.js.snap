// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseActivityScreen snapshot renders ChooseActivityScreen correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff7ae",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
        "height": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexGrow": 1,
        }
      }
    />
    <View
      style={
        Object {
          "height": 66.7,
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingHorizontal": 20,
              "paddingVertical": 30,
            },
            Object {
              "flexDirection": "row",
            },
            Object {
              "paddingVertical": 0,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 8,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.75)",
                "borderRadius": 5,
                "height": 10,
                "opacity": 0.5,
                "transform": Array [
                  Object {
                    "scale": 0.5,
                  },
                ],
                "width": 10,
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 8,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.75)",
                "borderRadius": 5,
                "height": 10,
                "opacity": 0.5,
                "transform": Array [
                  Object {
                    "scale": 0.5,
                  },
                ],
                "width": 10,
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 8,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.75)",
                "borderRadius": 5,
                "height": 10,
                "opacity": 0.5,
                "transform": Array [
                  Object {
                    "scale": 0.5,
                  },
                ],
                "width": 10,
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 8,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.75)",
                "borderRadius": 5,
                "height": 10,
                "opacity": 0.5,
                "transform": Array [
                  Object {
                    "scale": 0.5,
                  },
                ],
                "width": 10,
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 8,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.75)",
                "borderRadius": 5,
                "height": 10,
                "opacity": 0.5,
                "transform": Array [
                  Object {
                    "scale": 0.5,
                  },
                ],
                "width": 10,
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 8,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.75)",
                "borderRadius": 5,
                "height": 10,
                "opacity": 0.5,
                "transform": Array [
                  Object {
                    "scale": 0.5,
                  },
                ],
                "width": 10,
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 8,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.75)",
                "borderRadius": 5,
                "height": 10,
                "opacity": 0.5,
                "transform": Array [
                  Object {
                    "scale": 0.5,
                  },
                ],
                "width": 10,
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 8,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.75)",
                "borderRadius": 5,
                "height": 10,
                "opacity": 0.5,
                "transform": Array [
                  Object {
                    "scale": 0.5,
                  },
                ],
                "width": 10,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
